.github/workflows/build.yml

```diff
@@ -31,28 +31,31 @@ jobs:
       - name: Install Python
         uses: actions/setup-python@v5
         with:
-          python-version: "3.11"
+          python-version: "3.12"
+
+      - uses: astral-sh/setup-uv@v5
+        with:
+            enable-cache: true
+
       - name: Install dependencies
-        run: |
-          python -m pip install --upgrade pip
-          pip install ruff
+        run: uv sync --dev --no-install-package datafusion
```

Comment:
Nit: maybe just add a comment here saying that `--no-install-dependencies` is to only install the dependencies but not build the rust library